### PR TITLE
Merging the changes for elast-search-unit by switching to latest branch.

### PR DESCRIPTION
**Pull request by amolnin**
Merge request along with all the requestsed changes

### CODE REVIEW

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 1]
```diff
@@ -4,687 +4,902 @@
...
+		mockRulesForDelete();
+		
+		mockRulesForIndexes();
+		mockRulesHttpRequest();
+		mockRulesBulkInsert();
+
+	}
+	
+	@Test
+	public void testConnectionSuccess() {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Not a proper test scenario.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 2]
```diff
@@ -4,687 +4,902 @@
...
+		TransportClient client = mock(TransportClient.class);
+		assertNotNull(client);
+	}
+
+	@Test
+	public void testCreateIndexSuccess() {
+		boolean response = ElasticSearchUtil.createIndex(INDEX_NAME, TYPE_NAME, ElasticSearchMapping.createMapping(),
+				ElasticSearchSettings.createSettingsForIndex());
+		try {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Sleep not expected in the test.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 3]
```diff
@@ -4,687 +4,902 @@
...
+
+	@Test
+	public void testCreateDataSuccess() {
+
+		String responseId = ElasticSearchUtil.createData(INDEX_NAME, TYPE_NAME, (String) chemistryMap.get("courseId"), chemistryMap);
+		// inserting second record
+		ElasticSearchUtil.createData(INDEX_NAME, TYPE_NAME, (String) physicsMap.get("courseId"), physicsMap);
+		assertEquals(responseId, chemistryMap.get("courseId"));
+		// wait for 1 second since the created data will reflect after some time
+		try {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Sleep not expected in the test.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 4]
```diff
@@ -4,687 +4,902 @@
...
+		}
+		assertTrue(response);
+	}
+
+	@Test
+	public void testCreateDataSuccess() {
+
+		String responseId = ElasticSearchUtil.createData(INDEX_NAME, TYPE_NAME, (String) chemistryMap.get("courseId"), chemistryMap);
+		// inserting second record
+		ElasticSearchUtil.createData(INDEX_NAME, TYPE_NAME, (String) physicsMap.get("courseId"), physicsMap);
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: No check on second record inserted.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 5]
```diff
@@ -4,687 +4,902 @@
...
+
+	@Test
+	public void testGetByIdentifierSuccess() {
+		Map<String, Object> responseMap = ElasticSearchUtil.getDataByIdentifier(INDEX_NAME, TYPE_NAME,
+				(String) chemistryMap.get("courseId"));
+		assertEquals(responseMap.get("courseId"), chemistryMap.get("courseId"));
+	}
+
+	@Test
+	public void testUpdateByIdentifierSuccess() {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Rename to testUpdateDataSuccess

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 6]
```diff
@@ -4,687 +4,902 @@
...
+		GetResponse getResponse = mock(GetResponse.class);
+		when(client.prepareGet(Mockito.anyString(), Mockito.anyString(), Mockito.eq((String) chemistryMap.get("courseId"))))
+				.thenReturn(grb);
+		when(grb.get()).thenReturn(getResponse);
+		when(getResponse.getSource()).thenReturn(innermap);
+		
+		
+		boolean response = ElasticSearchUtil.updateData(INDEX_NAME, TYPE_NAME, (String) chemistryMap.get("courseId"), innermap);
+		assertTrue(response);
+		Map<String, Object> responseMap = ElasticSearchUtil.getDataByIdentifier(INDEX_NAME, TYPE_NAME,
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Why are we combining two API calls in same test? Remove code for ElasticSearchUtil.getDataByIdentifier.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 7]
```diff
@@ -4,687 +4,902 @@
...
+
+	@Test
+	public void testComplexSearchSuccess() throws Exception {
+		SearchDTO searchDTO = new SearchDTO();
+		List<String> fields = new ArrayList<String>();
+		fields.add("courseId");
+		fields.add("courseType");
+		fields.add("createdOn");
+		fields.add("description");
+		List<String> excludefields = new ArrayList<String>();
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: excludeFields

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 8]
```diff
@@ -4,687 +4,902 @@
...
+		searchDTO.setSoftConstraints(constraintMap);
+		searchDTO.setQuery("organisation Name published");
+		mockRulesForSearch(3);
+		Map map = ElasticSearchUtil.complexSearch(searchDTO, INDEX_NAME, TYPE_NAME);
+
+		assertEquals(3, map.size());
+	}
+
+	@Test
+	public void testComplexSearchWithRangeGreaterThanSuccess() {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: testComplexSearchSuccessWithRangeGreaterThan

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 9]
```diff
@@ -4,687 +4,902 @@
...
+		searchDTO.setAdditionalProperties(additionalProperties);
+		searchDTO.setQuery("organisation");
+		mockRulesForSearch(3);
+
+		Map map = ElasticSearchUtil.complexSearch(searchDTO, INDEX_NAME, TYPE_NAME);
+		assertEquals(2, map.size());
+	}
+
+	@Test
+	public void testComplexSearchWithRangeLessThanSuccess() {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: testComplexSearchSuccessWithRangeLessThan

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 10]
```diff
@@ -4,687 +4,902 @@
...
+
+		searchDTO.setAdditionalProperties(additionalProperties);
+		searchDTO.setQuery("organisation");
+		mockRulesForSearch(3);
+		Map map = ElasticSearchUtil.complexSearch(searchDTO, INDEX_NAME, TYPE_NAME);
+		assertEquals(2, map.size());
+	}
+
+	@Test
+	public void testGetByIdentifierWithoutIndexFailure() {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Likewise do for all tests ...

testGetByIdentifierFailureWithoutIndex

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 11]
```diff
@@ -4,687 +4,902 @@
...
+	}
+
+	@Test
+	public void testGetByIdentifierWithoutTypeAndIndexIdentifierFailure() {
+		Map<String, Object> responseMap = ElasticSearchUtil.getDataByIdentifier(null, null, "");
+		assertEquals(0, responseMap.size());
+	}
+
+	@Test
+	public void testGetDataWithoutIdentifierFailure() {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: testGetDataByIdentifierFailureWithoutIdentifier

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 12]
```diff
@@ -4,687 +4,902 @@
...
+		Map<String, Object> responseMap = ElasticSearchUtil.getDataByIdentifier(INDEX_NAME, TYPE_NAME, "");
+		assertEquals(0, responseMap.size());
+	}
+
+	@Test
+	public void testUpdateDataWithutIdentifierFailure() {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: testUpdateDataFailureWithoutIdentifier

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 13]
```diff
@@ -4,687 +4,902 @@
...
+	public void testGetByIdentifierSuccess() {
+		Map<String, Object> responseMap = ElasticSearchUtil.getDataByIdentifier(INDEX_NAME, TYPE_NAME,
+				(String) chemistryMap.get("courseId"));
+		assertEquals(responseMap.get("courseId"), chemistryMap.get("courseId"));
+	}
+
+	@Test
+	public void testUpdateByIdentifierSuccess() {
+		Map<String, Object> innermap = new HashMap<>();
+		innermap.put("courseName", "updatedCourese name");
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Updated course name

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 14]
```diff
@@ -4,687 +4,902 @@
...
+	@Test
+	public void testGetDataWithoutIdentifierFailure() {
+		Map<String, Object> responseMap = ElasticSearchUtil.getDataByIdentifier(INDEX_NAME, TYPE_NAME, "");
+		assertEquals(0, responseMap.size());
+	}
+
+	@Test
+	public void testUpdateDataWithutIdentifierFailure() {
+		Map<String, Object> innermap = new HashMap<>();
+		innermap.put("courseName", "updatedCourese name");
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Updated course name

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 15]
```diff
@@ -4,687 +4,902 @@
...
+		boolean response = ElasticSearchUtil.upsertData(null, TYPE_NAME, (String) chemistryMap.get("courseId"), innermap);
+		assertFalse(response);
+	}
+
+	@Test
+	public void testUpsertDataWithoutIndexTypeFailure() {
+		Map<String, Object> innermap = new HashMap<>();
+		innermap.put("courseName", "updatedCourese name");
+		innermap.put("organisationId", "updatedOrgId");
+		boolean response = ElasticSearchUtil.upsertData(null, null, (String) chemistryMap.get("courseId"), innermap);
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Index name shouldn't be null.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 16]
```diff
@@ -4,687 +4,902 @@
...
+				null);
+		assertFalse(response);
+	}
+
+
+
+	
+
+	@Test
+	public void testSaveDataWithoutIndexNameFailure() {
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: testCreateDataFailureWithoutIndexName

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 17]
```diff
@@ -4,687 +4,902 @@
...
+	}
+
+	@Test
+	public void testSearchDtoQueryTestSuccess() {
+		SearchDTO dto = new SearchDTO();
+		List<Map<String, Object>> list = new ArrayList<>();
+		Map<String, Object> map = new HashMap<>();
+		map.put("name", "test1");
+		list.add(map);
+		dto.setGroupQuery(list);
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Not a proper test scenario.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java [hunk 18]
```diff
@@ -4,687 +4,902 @@
...
+		assertTrue(dto.isFuzzySearch());
+	}
+
+	@Test
+	public void testSearchDtoArgumentedConstSuccess() {
+		List<Map> list = new ArrayList<>();
+		Map map = new HashMap();
+		map.put("name", "test1");
+		list.add(map);
+		SearchDTO dto = new SearchDTO(list, "add", 5);
```
Comment (bvinayakumar) [inline elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Not a proper test scenario.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java (hunk 6 above)
Comment (amolnin) [reply elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: The test-case is for update, so once create is done, next API call is for verifying the data by calling the getByIdentifier method.
Earlier - there was dependency created between the two test-cases, this was done to remove the dependency.

File: elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java (hunk 7 above)
Comment (amolnin) [reply elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java]: Did not understand? what action is required.

**Pull request merged**